File: src/test/java/com/testexercise/ebayframework/ShoppingCartPage.java
package com.testexercise.ebayframework;

import java.util.*;
import org.openqa.selenium.*;

public class ShoppingCartPage extends BasePage
{
    public ShoppingCartPage(AutomationFramework af)
    {
        super(af);
    }
    
    private final String shoppingCartListId = "ShopCart";
    
    public WebElement shoppingCartList()
    {
        return af.driver().findElement(By.id(shoppingCartListId));
    }
    
    private final String cartItemInfoCss = "div.fr.itemInfoColcart140";
    
    public List<ItemDetails> getItemsInCart()
    {
        List<WebElement> cartItems = shoppingCartList().
                findElements(By.cssSelector(cartItemInfoCss));
        
        ArrayList<ItemDetails> details =
                new ArrayList<ItemDetails>(cartItems.size());
        
        for (WebElement cartItem: cartItems)
        {
            details.add(new CartItemWidget(cartItem).getItemDetails());
        }
        
        return details;
    }
}

File: net/TestExerciseEbay/EbayPOMFramework/ShoppingCartPage.cs
using System;
using System.Collections.Generic;
using OpenQA.Selenium;

namespace TestExercise.EbayPOMFramework
{
    public class ShoppingCartPage : BasePage
    {
        public ShoppingCartPage(AutomationFramework af) : base(af) { }

        private const String shoppingCartListId = "ShopCart";

        public IWebElement ShoppingCartList
        {

            get { return af.Driver.FindElement(By.Id(shoppingCartListId)); }
        }

        private const string cartItemInfoCss = "div.fr.itemInfoColcart140";

        public IList<ItemDetails> GetItemsInCart()
        {
            IReadOnlyCollection<IWebElement> cartItems =
                    ShoppingCartList.FindElements(By.CssSelector(cartItemInfoCss));

            List<ItemDetails> details =
                    new List<ItemDetails>(cartItems.Count);

            foreach (IWebElement cartItem in cartItems)
            {
                details.Add(new CartItemWidget(cartItem).GetItemDetails());
            }

            return details;
        }
    }
}
